applications/dairdrop.md

```diff
@@ -0,0 +1,232 @@
+
+This community has continually considered and sought solutions to minimize expenses for a decentralized project, which has been a persistent goal of the Polkadot community. In the past, when some large token holders in the Dota community wanted to airdrop to community users, they felt frustrated because there was no simple tool to help them realize this idea.
+
+These challenges have hindered the development of decentralized projects. Polkadot has great technology, and community developers should take steps to lower the usage barriers for users in various aspects, making it accessible to more ordinary people not just a tool for technical experts and the wealthy.
+
+This application is designed for community projects that frequently need to conduct airdrops. It allows anyone to airdrop to their community or claim their own airdrop without relying on centralized servers, reducing dependence on centralized teams. 
+
+It can also mitigate the issue of airdropping to inactive accounts, as we know that airdropping to inactive accounts wastes valuable on-chain resources and can lead to permanent losses in on-chain liquidity. This phenomenon was particularly evident in DED's past MEME activities. Most importantly, it can unify airdrop standards on-chain, allowing users to monitor community airdrop activities based on on-chain transaction event information, thus improving transparency and enabling more users to participate in a timely manner.
+
+### Project Details
+
+- language `rust`（tauri）、`javascript`、`python`
+  > It will use the Tauri framework to achieve compatibility across multiple platforms, allowing it to be used on Windows, Linux, and macOS
+### Features
+
+- **Desktop APP**
+  - Connect to RPC nodes and create or connect wallets
+  - Support registering airdrop information on-chain (MD5 of the airdrop file, root of the Merkle tree for the airdrop accounts and amounts, start block, duration, URL of the airdrop file)
+  - Airdrop list (displaying airdrop status) and query airdrop status by user address
+  - Claim airdrops for any user (bulk claiming supported)
+  - Store airdrop data using Merkle tree
+  - Compatible with multiple operating systems(Linux、 Mac、 windows)
+
+- **Web Service**
+  - A web service with all the above functionalities
+
+- **An airdrop indexer**
+
+- **On-chain**
+  - An airdrop pallet(substrate pallet)
+    - Set airdrop: Merkle tree root, MD5 hash of the airdrop file, start date, end date, total airdrop amount (must be staked)
```

Comment:
To what granularity will the merkle tree resolve? To each individual address? To a group of addresses which will always be claimed together? I'd love to hear more about the strategy. Have you conducted any research up to which address amount your solution is able to scale?

Reply:
> To what granularity will the merkle tree resolve? To each individual address? To a group of addresses which will always be claimed together? I'd love to hear more about the strategy. Have you conducted any research up to which address amount your solution is able to scale?

We will provide a complete Merkle tree for each airdrop, which includes the airdrop information for every user and is stored in the backend. This will support up to 2^160 addresses.

Reply:
If you are storing all the addresses individually why do you need the merkle tree at all? Still do you have a more in depth design doc to share about your concept? I'd appreciate it.

Reply:
> If you are storing all the addresses individually why do you need the merkle tree at all? Still do you have a more in depth design doc to share about your concept? I'd appreciate it.

You can think of each airdrop event as a Merkle tree, where each person's airdrop information is stored in the leaves. They will receive their Merkle proof from here and then submit it to the chain for verification to obtain their airdrop amount.

Reply:
This is very flexible and scalable. By adding some information to each recipient's airdrop data, we can create conditional claims for everyone, which can be applied in gaming and social applications.

Reply:
I'm still keen to see a design document